scripts/image-builder/build.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+set -e -o pipefail
+
+IMGNAME=microshift
+ROOTDIR=$(git rev-parse --show-toplevel)/scripts/image-builder
+
+trap ${ROOTDIR}/cleanup.sh INT
+
+title() {
+    echo -e "\E[34m\n# $1\E[00m";
+}
+
+waitfor_image() {
+    local uuid=$1
+
+    local tstart=$(date +%s)
+    echo "$(date +'%Y-%m-%d %H:%M:%S') STARTED"
+
+    local status=$(sudo composer-cli compose status | grep ${uuid} | awk '{print $2}')
+    while [ "${status}" = "RUNNING" ]; do
+        sleep 10
+        status=$(sudo composer-cli compose status | grep ${uuid} | awk '{print $2}')
+        echo -en "$(date +'%Y-%m-%d %H:%M:%S') ${status}\r"
+    done
+
+    local tend=$(date +%s)
+    echo "$(date +'%Y-%m-%d %H:%M:%S') ${status} - elapsed $(( (tend - tstart) / 60 )) minutes"
+    
+    if [ "${status}" = "FAILED" ]; then
+        download_image ${uuid} 1
+        echo "Blueprint build has failed. For more information, review the downloaded logs"
+        exit 1
+    fi
```

Comment:
This contains good improvements over `https://github.com/redhat-et/microshift-demos/blob/main/_lib/build.sh`. Would it make sense to contribute them back there, too?